Increase provisioning test timeouts.

jsafrane opened the pull request:
We've encountered flakes in our e2e infrastructure when kubelet took more than one minute to detach a volume used by a deleted pod.

Let's increase the wait period from 1 to 3 minutes. This slows down the test by 2 minutes, but it makes the test more stable.

In addition, when kubelet cannot detach a volume for 3 minutes, let the test wait for additional recycle controller retry interval (10 minutes) and hope the volume is deleted by then. This should not increase usual test time, it makes the test stable when kubelet is _extremely_ slow when releasing the volume.

Fixes: #24161

jsafrane commented:
@saad-ali, @rootfs PTAL
cc: @kubernetes/sig-storage

rootfs commented on test/e2e/volume_provisioning.go:
```diff
@@ -109,16 +109,16 @@ var _ = KubeDescribe("Dynamic provisioning", func() {
+			// We've seen GCE PD detach to take more than 1 minute.
 			By("Sleeping to let kubelet destroy all pods")
-			time.Sleep(time.Minute)
+			time.Sleep(3 * time.Minute)
```
is it possible to use `waitForPodTerminated` to monitor pods termination instead of sleep?

jsafrane replied:
No, it is not. As you can see few lines above, the test *does* wait for pod termination. And as described in the comment, when a pod is deleted all its volumes are still attached to the node and kubelet (slowly) starts detaching them. There is no event that would signalize that kubelet finished and all volumes are detached.

saad-ali replied:
You could get fancy and retrieve the volume name from the PV object and  use it to figure out what node it gets attached to and monitor for detach. See https://github.com/kubernetes/kubernetes/blob/master/test/e2e/pd.go#L498

jsafrane replied:
Well, I _could_, but as you can notice, AWS and Cinder is not implemented there.  Common cloud volume itnerface as suggested in https://github.com/kubernetes/kubernetes/issues/21984 would help a lot (with `IsDiskAttached()` method).

I'd propose fixing the flake now and polishing the timers later.

saad-ali replied:
Good point, agreed.

jsafrane committed:
Increase provisioning test timeouts.

We've encountered flakes in our e2e infrastructure when kubelet took more than
one minute to detach a volume used by a deleted pod.

Let's increase the wait period from 1 to 3 minutes. This slows down the test
by 2 minutes, but it makes the test more stable.

In addition, when kubelet cannot detach a volume for 3 minutes, let the test
wait for additional recycle controller retry interval (10 minutes) and hope the
volume is deleted by then. This should not increase usual test time, it makes
the test stable when kubelet is _extremely_ slow when releasing the volume.

jsafrane commented:
rebased

saad-ali commented:
LGTM
